### DISCUSSION TITLE
Add brainageR v2.1

### DISCUSSION THREAD

**Discussion by JavierSanchez3** [neurodesk/neurodesk.github.io-hugo#428] [New Tool]
Hello!

Is it possible to add brainageR v2.1?  

https://github.com/james-cole/brainageR

Best regards!

**Comment by stebo85** [comment]
Dear @JavierSanchez3,

Yes, that's possible :) You could either write a new recipe for brainageR and propose it to our neurocontainers repo (e.g. based on https://github.com/fprados/brainageR_dockerfile ) or you build the brainageR_dockerfile, then upload the container to dockerhub and then use a simple recipe similar to fMRIprep (https://github.com/NeuroDesk/neurocontainers/blob/master/recipes/fmriprep/build.sh)

Can you give this is a go and let us know if you need any help?

Thank you
Steffen

**Reply by JavierSanchez3** [reply]
Hello Steffen,

I tried to create the build.sh script from the Dockerfile; however, an error appeared (see the attached document). 

[build_sh_attempt.txt](https://github.com/user-attachments/files/16253572/build_sh_attempt.txt)

Do you recommend that I edit the original Dockerfile to try to correctly generate the build.sh script? In addition, I made another attempt by editing the Dockerfile, but further errors appeared when trying to install FSL 5.0.

Thanks for your support.

Best regards

**Reply by stebo85** [reply]
yes, these issues can be challenging and one option is to google the errors and troubleshoot them step by step, 

e.g. libdeflate-dev 

you can find out here https://pkgs.org/search/?q=libdeflate-dev  which distros have this package - there you see that you could try ubuntu 20.04 as a base image instead of 16.04

**Comment by neurolabusc** [comment]
@stebo85 I have written a Python script for using brainager on neurodesk. It is pretty amazing - traditionally I have run brainager sequentially one individual at a time, but thanks to mcr on neurodesk I can run 96 instances at a time on my Threadripper, using a simple wrapper to read an entire BIDS folder. Neurodesk is really a game changer. 

I think my script would be useful to others, but it will need integration from the neurodesk team. I tried to use the original brainager that has dependencies for spm12, Rscript, and fsl. The challenge is that "ml fsl" includes a huge number of tis own libraries and seems to wipe out Rscript. For this reason, my current script skips the creation of QA images (though this can be done in a separate pass). 

If you email me, I can send you a link to my repository. It includes instructions and a sample T1 scan to demonstrate.

Once again, I am in awe of neurodesk. It really helps bring neuroimaging tools to scale.

**Reply by stebo85** [reply]
Great - will do this once i am back to work mid October

**Reply by stebo85** [reply]
@neurolabusc - can we look at this together next week and get this all nicely working?

**Comment by stebo85** [comment]
that's completed.
